src/python/pants/engine/target.py

```diff
@@ -2466,10 +2477,31 @@ def unevaluated_transitive_excludes(self) -> UnparsedAddressInputs:
         )
 
 
+# If this predicate returns false, then the target's field's deps will be ignored.
+ShouldTraverseDepsPredicate = Callable[[Target, Field], bool]
```

Comment:
Yeah. I tried that first, but ran into issues with the order those are defined. Because this is not just a hint, but a definition, `Dependencies` and `SpecialCasedDependencies` have to be defined earlier in the file. I was trying to keep the type definition and the functions close to where they are used.

Reply:
Can you use string annotations and/or `if TYPE_CHECKING`?

Reply:
Addressed in #19272